kola/harness.go

```diff
@@ -201,6 +201,24 @@ func filterTests(tests map[string]*register.Test, pattern, platform string, vers
 			continue
 		}
 
+		allowed = true
```

Comment:
since this is pretty much the same as the platform code, we could generalize and break out into a new function